src/week-view-loader.ts

```diff
@@ -0,0 +1,101 @@
+"use strict";
+
+let weekOffset: number = 0;
+
+async function loadWeekView(){
+    loadCalendarGrid(7);
+    loadTimezoneLabel();
+    await reloadWeekView();
+}
+
+async function reloadWeekView(){
+    loadCalendarDateLabels();
+    await loadEvents();
+    loadHeaderDate(getOffsetDate());
+}
+
+function getOffsetDate(){
+    return new Date(LOAD_TIME.getTime() + weekOffset * 7 * TIME_IN_A_DAY_MS);
+}
+
+function loadTimezoneLabel(){
+    let timezoneElement = document.getElementsByClassName("week-view__dates-header-timezone")[0] as HTMLElement;
+    let timezoneOffset = LOAD_TIME.toString().split("GMT")[1].slice(0, 3);
+    timezoneElement.innerText = `UTC${timezoneOffset}`;
+}
+
+function loadCalendarDateLabels(){
+    const firstDayOfTheWeek: Date = getOffsetDate().getFirstDayOfWeek();
+
+    let elements = document.getElementsByClassName("week-view__dates-header-date");
+    let dateColumns = document.getElementsByClassName("calendar-grid__calendar-column");
+    for (let i = 0; i < elements.length; i++) {
+        let element = elements[i];
+        const elementWeekdayLabel: HTMLElement = element.getElementsByClassName("calendar-date-weekday")[0] as HTMLElement;
+        const elementDateLabel: HTMLElement = element.getElementsByClassName("calendar-date-day")[0] as HTMLElement;
+        elementWeekdayLabel.innerText = THREE_LETTER_WEEK_DAYS[firstDayOfTheWeek.getDay()];
+        elementDateLabel.innerText = firstDayOfTheWeek.getDate().toString();
+
+        const date: string = `${firstDayOfTheWeek.getFullYear()}-${firstDayOfTheWeek.getMonth() + 1}-${firstDayOfTheWeek.getDate()}`;
+        dateColumns[i].setAttribute("data-calendar-day", date);
+
+        if (firstDayOfTheWeek.toDateString() === LOAD_TIME.toDateString()){
+            element.classList.add("week-view__dates-header-date--active");
+        }
+        else element.classList.remove("week-view__dates-header-date--active");
+
+        firstDayOfTheWeek.setDate(firstDayOfTheWeek.getDate() + 1);
+    }
+}
+
+function loadCalendarGrid(columnCount: number) {
+    const calendarGrid: HTMLElement = document.getElementsByClassName("week-view__calendar-grid")[0] as HTMLElement;
+
+    const calendarGridTimeColumn: HTMLElement = document.createElement("section");
+    calendarGridTimeColumn.classList.add("calendar-grid__time-column");
+    for (let i = 1; i <= 24; i++){
+        const calendarGridTimeColumnCell: HTMLElement = document.createElement("div");
+        calendarGridTimeColumnCell.classList.add("time-column__cell");
+        calendarGridTimeColumnCell.innerText = i===24? "" : `${i.leftPad(2)}:00`;
+        calendarGridTimeColumn.appendChild(calendarGridTimeColumnCell);
+    }
+    calendarGrid.appendChild(calendarGridTimeColumn);
+    for (let i = 0; i < columnCount; i++){
+        const calendarGridColumn: HTMLElement = document.createElement("section");
+        calendarGridColumn.classList.add("calendar-grid__calendar-column");
+        for (let j = 0; j < 24; j++){
```

Comment:
24 here would be a magic number, try looking the term up in the internet and see what are the downsides